src/routes/groups/groups.controller.js

```diff
@@ -0,0 +1,60 @@
+const groupsData = require('./../../../db/groups.data.json')
+const fs = require("fs");
+const { promisify } = require("util");
+
+const writeFile = promisify(fs.writeFile);
+
+const listGroup = (req, res) => {
+  return res.json(groupsData);
+};
+
+const postGroup = async (req, res) => {
+  const id = groupsData.groups.length + 1;
+  const newgroupsData = {
+    groups: [...groupsData.groups, { id: parseInt(id, 10), ...req.body }]
+  };
+  await writeFile("db/groups.data.json", JSON.stringify(newgroupsData));
+  res.status(201);
+  return res.json({
+    ...req.body
+  });
+};
+
+const updateGroup = async (req, res) => {
+  const groupsArray = groupsData.groups
+  const updatedGroupsArray = groupsArray.map((group) => {
+    if (group.id === parseInt(req.params.groupId, 10)) {
+      return {...group, ...req.body}
+    }
+    return {...group}
+  }) 
```

Comment:
How might you redesign the app so that looking up a group record doesn't require searching an array? (hint: think data structures)